lms/djangoapps/courseware/views/views.py

```diff
@@ -439,36 +448,79 @@ def get_last_accessed_courseware(course, request, user):
     return None
 
 
-@ensure_csrf_cookie
-@ensure_valid_course_key
-def static_tab(request, course_id, tab_slug):
+class StaticCourseTabView(FragmentContainerView):
     """
-    Display the courses tab with the given name.
-
-    Assumes the course_id is in a valid format.
+    View that displays a static course tab with a given name.
     """
+    @method_decorator(ensure_csrf_cookie)
+    @method_decorator(ensure_valid_course_key)
+    def get(self, request, course_id, tab_slug, **kwargs):
+        """
+        Displays a static course tab page with a given name
+        """
+        course_key = CourseKey.from_string(course_id)
+        course = get_course_with_access(request.user, 'load', course_key)
+        tab = CourseTabList.get_tab_by_slug(course.tabs, tab_slug)
+        if tab is None:
+            raise Http404
+        return self.render_fragment_to_response(request, course=course, tab=tab, **kwargs)
 
-    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    def render_fragment(self, request, course=None, tab=None, **kwargs):
+        """
+        Renders the course tab's fragment.
+        """
+        return get_static_tab_fragment(request, course, tab)
 
-    course = get_course_with_access(request.user, 'load', course_key)
+    def render_fragment_to_string(self, request, fragment, course=None, tab=None, **kwargs):
+        """
+        Renders the fragment to a string.
+        """
+        return render_to_response('courseware/static_tab.html', {
+            'course': course,
+            'active_page': 'static_tab_{0}'.format(tab['url_slug']),
+            'tab': tab,
+            'fragment': fragment,
+            'uses_pattern_library': False,
+            'disable_courseware_js': True,
+        })
 
-    tab = CourseTabList.get_tab_by_slug(course.tabs, tab_slug)
-    if tab is None:
-        raise Http404
 
-    contents = get_static_tab_contents(
-        request,
-        course,
-        tab
-    )
-    if contents is None:
-        raise Http404
+class CourseTabView(FragmentContainerView):
+    """
+    View that displays a course tab page.
+    """
+    @method_decorator(ensure_csrf_cookie)
+    @method_decorator(ensure_valid_course_key)
+    def get(self, request, course_id, tab_type, **kwargs):
+        """
+        Displays a course tab page that contains a web fragment.
+        """
+        course_key = CourseKey.from_string(course_id)
+        course = get_course_with_access(request.user, 'load', course_key)
+        tab = [tab for tab in course.tabs if tab.type == tab_type][0]
```

Comment:
@andy-armstrong might be error prone - what if there are no matching tab?

Maybe something like

```
try:
    tab = next(tab for tab in course.tabs if tab.type == tab_type)
except StopIteration:
    tab = None
```

would work better?